core-data/resolvers.js

```diff
@@ -17,3 +17,14 @@ export async function* getCategories() {
 	const categories = await apiRequest( { path: '/wp/v2/categories' } );
 	yield receiveTerms( 'categories', categories );
 }
+
+/**
+ * Requests a media element from the REST API.
+ *
+ * @param {Object} state State tree
+ * @param {number} id    Media id
+ */
+export async function* getMedia( state, id ) {
```

Comment:
I was wondering if passing the `state` to the resolvers is a good idea? Maybe it's useless, since we can still call `select` from inside the resolver if needed. Kept it as is for now, but let's see how this goes.

Reply:
Indeed, I'm not too sure how much we want to promote the use of `select`, particularly in a context where we know we could call selectors via direct import and pass the state. Another reason was merely to align the signatures between selectors and resolvers (since their names already match).

See also: https://github.com/WordPress/gutenberg/pull/5219#issuecomment-372184264